re-evaluate testing different combinations of C-stdlibs, C++-stdlibs and C++-rtlibs

cmake flags = -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/where/i/am/going

The CMakeError.log file contains:
> Performing C++ SOURCE FILE Test CXX14_BUILTIN failed with the following output:
> Change Dir: /shares/apps/seqan/build/CMakeFiles/CMakeTmp
> 
> Run Build Command:"/usr/bin/gmake" "cmTC_31c4b/fast"
> /usr/bin/gmake -f CMakeFiles/cmTC_31c4b.dir/build.make CMakeFiles/cmTC_31c4b.dir/build
> gmake[1]: Entering directory `/shares/apps/seqan/build/CMakeFiles/CMakeTmp'
> Building CXX object CMakeFiles/cmTC_31c4b.dir/src.cxx.o
> /apps/intel/2017/compilers_and_libraries_2017.0.098/linux/bin/intel64/icpc     -DCXX14_BUILTIN   -o CMakeFiles/cmTC_31c4b.dir/src.cxx.o -c /shares/apps/seqan/build/CMakeFiles/CMakeTmp/src.cxx
> /shares/apps/seqan/build/CMakeFiles/CMakeTmp/src.cxx(2): error: #error directive: NOCXX14
>       #error NOCXX14
>        ^
> 
> compilation aborted for /shares/apps/seqan/build/CMakeFiles/CMakeTmp/src.cxx (code 2)
> gmake[1]: *** [CMakeFiles/cmTC_31c4b.dir/src.cxx.o] Error 2
> gmake[1]: Leaving directory `/shares/apps/seqan/build/CMakeFiles/CMakeTmp'
> gmake: *** [cmTC_31c4b/fast] Error 2
> 
> Source file was:
> #if !defined(__cplusplus) || (__cplusplus < 201300L)
>     #error NOCXX14
>     #endif
>     int main() {}
> Performing C++ SOURCE FILE Test CXX14_FLAG failed with the following output:
> Change Dir: /shares/apps/seqan/build/CMakeFiles/CMakeTmp
> 
> Run Build Command:"/usr/bin/gmake" "cmTC_46dbd/fast"
> /usr/bin/gmake -f CMakeFiles/cmTC_46dbd.dir/build.make CMakeFiles/cmTC_46dbd.dir/build
> gmake[1]: Entering directory `/shares/apps/seqan/build/CMakeFiles/CMakeTmp'
> Building CXX object CMakeFiles/cmTC_46dbd.dir/src.cxx.o
> /apps/intel/2017/compilers_and_libraries_2017.0.098/linux/bin/intel64/icpc     -std=c++14 -DCXX14_FLAG   -o CMakeFiles/cmTC_46dbd.dir/src.cxx.o -c /shares/apps/seqan/build/CMakeFiles/CMakeTmp/src.cxx
> /shares/apps/seqan/build/CMakeFiles/CMakeTmp/src.cxx(2): error: #error directive: NOCXX14
>       #error NOCXX14
>        ^
> 
> compilation aborted for /shares/apps/seqan/build/CMakeFiles/CMakeTmp/src.cxx (code 2)
> gmake[1]: *** [CMakeFiles/cmTC_46dbd.dir/src.cxx.o] Error 2
> gmake[1]: Leaving directory `/shares/apps/seqan/build/CMakeFiles/CMakeTmp'
> gmake: *** [cmTC_46dbd/fast] Error 2
> 
> Source file was:
> #if !defined(__cplusplus) || (__cplusplus < 201300L)
>     #error NOCXX14
>     #endif
>     int main() {}


If I compile `int main(){}` with icpc --std=c++14, i get no compile errors

I have no issues using gcc 6.2.0, but would like to use Intel for obvious reasons.  Any help would be greately appreciated.

--- a/manual/source/Infrastructure/Use/CustomBuildSystem.rst
+++ b/manual/source/Infrastructure/Use/CustomBuildSystem.rst
@@ -35,6 +35,31 @@ For XCode on macOS OpenMP is not yet available.
 
 With Visual Studio OpenMP is switched on by default.
 
+Compiler specifics
+------------------
+
+Intel Compiler
+^^^^^^^^^^^^^^
+
+The Intel Compiler does not ship a c++ standard library on its own and will use
+the one pre-installed on the system (e.g., the one from g++). This can be a
+problem [especially for cluster users through the use of a module system], if
+the standard library by a default g++ installation is to old.
+
+Please check with the following command which g++ version is being used and make
+sure it matches the supported gcc versions.
+
+.. code-block:: console
+
+    # icpc -v
+    icpc version 17.0.2 (gcc version 5.4.0 compatibility)
+
+If you have multiple g++ installations, you can choose the standard library by
+``icpc -gxx-name=g++-5.4.0 -gcc-name=gcc-5.4.0 …``.
+
+You may have to add the path of the library to ```$LD_LIBRARY_PATH`` for the
+linker.
+
 Operating System specifics
 --------------------------
 

--- a/manual/source/Infrastructure/Use/FindSeqAnCMake.rst
+++ b/manual/source/Infrastructure/Use/FindSeqAnCMake.rst
@@ -269,6 +269,30 @@ Required additions to C++ compiler flags are in the following variable:
 
     Please note that these variables include whatever has been added by the dependencies mentioned above so **do not add** e.g. ``${OpenMP_CXX_FLAGS}`` yourself!
 
+Intel Compiler specifics
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+The Intel Compiler does not ship a c++ standard library on its own and will use
+the one pre-installed on the system (e.g., the one from g++). This can be a
+problem [especially for cluster users through the use of a module system], if
+the standard library by a default g++ installation is to old.
+
+Please check with the following command which g++ version is being used and make
+sure it matches the supported gcc versions.
+
+.. code-block:: console
+
+    # icpc -v
+    icpc version 17.0.2 (gcc version 5.4.0 compatibility)
+
+If you have multiple g++ installations, you can choose the standard library by
+``icpc -gxx-name=g++-5.4.0 -gcc-name=gcc-5.4.0 …``. Use
+``cmake -DCMAKE_CXX_FLAGS="-gxx-name=g++-5.4.0 -gcc-name=gcc-5.4.0" …``
+to propagate those options through cmake.
+
+You may have to add the path of the library to ```$LD_LIBRARY_PATH`` for the
+linker.
+
 Static builds
 ^^^^^^^^^^^^^
 

